code/modules/projectiles/guns/projectile/pistol.dm

```diff
@@ -250,12 +249,47 @@
 		Fire(M,M)
 		return 1
 	return ..()
+	
+/obj/item/weapon/gun/projectile/glock/examine(mob/user)
+	..()
+	if(conversionkit)
+		to_chat(user, "<span class='info'>There's something screwed onto the back.</span>")
 
 /obj/item/weapon/gun/projectile/glock/fancy
 	name = "\improper NT Glock Custom"
 	desc = "The NT Glock is a cheap, ubiquitous sidearm, produced by a NanoTrasen subsidiary. Uses .380AUTO rounds. Its subcompact frame can fit in your pocket. <br><span class='notice'>This one has a sweet platinum-plated slide, and tritium night sights for maintenance crawling!</span>"
 	icon_state = "secglockfancy"
 	clowned = UNCLOWN
+	
+/obj/item/weapon/gun/projectile/glock/flame
+	name = "\improper NT Glock Custom"
+	desc = "The NT Glock is a cheap, ubiquitous sidearm, produced by a NanoTrasen subsidiary. Uses .380AUTO rounds. Its subcompact frame can fit in your pocket. <br><span class='notice'>This one has rad flame stickers on it!</span>"
+	icon_state = "secglockflame"
+	spawn_mag = FALSE
+	clowned = UNCLOWN
+	w_class = W_CLASS_MEDIUM
+	
+/obj/item/weapon/gun/projectile/glock/flame/New()	
+	..()
+	conversionkit = new /obj/item/gun_part/glock_auto_conversion_kit(src)
+	var/image/auto_overlay = image("icon" = 'icons/obj/gun_part.dmi', "icon_state" = "auto_attach")
```

Comment:
this is copypasted from the `attackby`. Why not put it in update_icon instead?

Reply:
Attackby isn't being called, and it only needs to be ran once (on attach or removal).